frontend/src/components/pages/search/Search.tsx

```diff
@@ -33,19 +36,43 @@ const Search: React.FC = () => {
     setName(e.target.value);
   };
 
-  const handleCheckbox = (e: React.ChangeEvent<HTMLInputElement>) => {
-    setCheckboxFilters({ ...checkboxFilters, [e.target.name]: e.target.checked });
+  const handleOnline = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setIsOnline({ ...online, [e.target.name]: e.target.checked });
+  };
+
+  const handleGender = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setGenders({ ...genders, [e.target.name]: e.target.checked });
   };
 
   const handleSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
     setSelectsFilters({ ...selectsFilters, [e.target.name]: e.target.value });
   };
 
-  const getGender = () => {
-    if ((!checkboxFilters.female && !checkboxFilters.male) || (checkboxFilters.female && checkboxFilters.male)) {
+  useEffect(() => {
+    async function getUsers() {
+      await apiService(dataType.users).then(users => setUsers(users));
```

Comment:
the point was to 

const user = await apiService(dataType.users);

setUsers(users)